frontend/packages/metal3-plugin/src/components/host-detail.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+
+import { Row, Col } from 'patternfly-react';
+
+import { referenceForModel } from '@console/internal/module/k8s';
+import { DetailsPage } from '@console/internal/components/factory';
+import { navFactory } from '@console/internal/components/utils';
+import { units } from '@console/internal/components/utils/units';
+import { getName } from '@console/shared';
+
+import { BaremetalHostModel } from '../models';
+import MachineCell from './machine-cell';
+import {
+  getHostNICs,
+  isHostOnline,
+  getHostDescription,
+  getHostBMCAddress,
+  getHostCPUs,
+  getHostRAM,
+  getHostTotalStorageCapacity,
+} from '../selectors';
+
+type BaremetalHostDetailPageProps = {
+  namespace: string;
+  name: string;
+  match: any;
+};
+
+const BaremetalHostDetails = ({ obj }) => {
+  const nics = getHostNICs(obj);
+  const online = isHostOnline(obj);
+  const ips = nics.map((nic) => nic.ip).join(', ');
+
+  const statusClasses = classNames({
+    fa: true,
+    'co-icon-and-text__icon': online === true,
+    'fa-refresh': true,
+  });
+
+  const totalCapacity = units.humanize(
+    // The value from the selector is in GB
+    getHostTotalStorageCapacity(obj) * 1024 ** 3,
+    'decimalBytes',
+    true,
+  ).string;
+
+  return (
+    <div className="co-m-pane__body">
+      <h1 className="co-m-pane__heading">Baremetal Host Overview</h1>
+      <Row>
+        <Col lg={4} md={4} sm={4} xs={4} id="name-description-column">
+          <dl>
+            <dt>Name</dt>
+            <dd>{getName(obj)}</dd>
+            <dt>Description</dt>
+            <dd>{getHostDescription(obj)}</dd>
+          </dl>
+        </Col>
+        <Col lg={3} md={3} sm={3} xs={3}>
+          <dl>
+            <dt>Status</dt>
+            <dd>
+              <span className="co-icon-and-text">
+                <span aria-hidden="true" className={statusClasses} />
+                {online ? 'Running' : 'Not running'}
+              </span>
+            </dd>
+            <dt>Management Address</dt>
+            <dd>{getHostBMCAddress(obj)}</dd>
+            <dt>IP Addresses</dt>
+            <dd>{ips}</dd>
+          </dl>
+        </Col>
+        <Col lg={3} md={3} sm={3} xs={3}>
+          <dl>
+            <dt>Machine</dt>
+            <dd>
+              <MachineCell host={obj} />
+            </dd>
+          </dl>
+        </Col>
+        <Col lg={2} md={2} sm={2} xs={2}>
+          <dl>
+            <dt>Hardware</dt>
+            <dd>{getHostCPUs(obj).length} CPU cores</dd>
+            <dd>{getHostRAM(obj)}GB RAM</dd>
```

Comment:
Same comment here on the helper. I'd suggest changing it to

```suggestion
            <dd>{getHostRAMGB(obj)}GB RAM</dd>
```